### PR TITLE
Add a clear_state for DataReportingFlags

### PR DESCRIPTION
**Pull request by jmarrec**
hotfix for new tests failing serially following #7970

### CODE REVIEW

File: src/EnergyPlus/DataReportingFlags.cc [hunk 1]
```diff
@@ -94,6 +94,21 @@ namespace DataReportingFlags {
...
+        NumOfWarmupDays = 0;
+        cWarmupDay.clear();
+        DisplayPerfSimulationFlag = false;
+        DoWeatherInitReporting = false;
+        PrintEndDataDictionary = false;
+        MakeMirroredDetachedShading = true;
+        MakeMirroredAttachedShading = true;
+        DebugOutput = false;
+        EvenDuringWarmup = false;
+    }
```
Comment (jmarrec) [inline src/EnergyPlus/DataReportingFlags.cc:110]: new clear_state for DataReportingFlags

File: src/EnergyPlus/StateManagement.cc [hunk 2]
```diff
@@ -262,6 +263,7 @@ void EnergyPlus::clearAllStates(OutputFiles &outputFiles)
     DataOutputs::clear_state();
     DataPhotovoltaics::clear_state();
     DataPlant::clear_state();
+    DataReportingFlags::clear_state();
```
Comment (jmarrec) [inline src/EnergyPlus/StateManagement.cc:266]: Register it in StateManagement

**Comment by Myoldmopar** [comment]
@mjwitte if you can confirm this fixes it locally for you then feel free to merge at will.  CI wasn't seeing the problem, so I wouldn't anticipate any effect on here.

**Comment by jmarrec** [comment]
FWIW I tested locally by running tests serially which failed before and worked after.

**Review by mjwitte — APPROVED** [review]
Tested locally with SimulationManager* tests, problem is fixed. Cautious me wants to wait for at least one round of CI results, if they show up soon.

**Comment by mjwitte** [comment]
No CI results yet, merging anyway.

**Pull request merged**